src/OrbitBase/File.cpp

```diff
@@ -76,6 +76,24 @@ ErrorMessageOr<void> WriteFully(const unique_fd& fd, std::string_view content) {
   return outcome::success();
 }
 
+ErrorMessageOr<void> WriteFullyAtOffset(const unique_fd& fd, const void* buffer, size_t size,
+                                        off_t offset) {
+  const char* current_position = static_cast<const char*>(buffer);
+
+  while (size > 0) {
+    int64_t bytes_written = TEMP_FAILURE_RETRY(pwrite(fd.get(), current_position, size, offset));
```

Comment:
CHECK(bytes_written != 0);

Reply:
Hmm, is this really an error?

_The number of bytes written may be less than count if, for example, there is insufficient space on the underlying physical medium, or the RLIMIT_FSIZE resource limit is encountered (see setrlimit(2)), or the call was interrupted by a signal handler after having written less than count bytes. (See also pipe(7).)_

If pwrite gets interrupted with zero bytes read we should retry, right?

Reply:
hmm, I assume that for write if 0 bytes are written it returns -1 with EITNR, at least this is what write does

_EINTR  The call was interrupted by a signal before any data was written; see signal(7)._

so it can never be 0.

Reply:
and EINTR is handled by TEMP_FAILURE_RETRY macro.

Reply:
> EINTR The call was interrupted by a signal before any data was written; see signal(7).

Okay, then the Check should be save. Added. Done.